helix-view/src/editor/variable_expansion.rs

```diff
@@ -67,6 +67,22 @@ impl Editor {
                                             .cursor_line(doc.text().slice(..))
                                             + 1)
                                         .to_string(),
+                                        "cursorcolumn" => (doc
+                                            .selection(view.id)
+                                            .primary()
+                                            .cursor(doc.text().slice(..))
+                                            + 1)
+                                        .to_string(),
+                                        "lang" => doc.language_name().unwrap_or("text").to_string(),
+                                        "ext" => match doc.relative_path() {
+                                            Some(path) => path
+                                                .to_string_lossy()
+                                                .split(".")
+                                                .last()
+                                                .unwrap_or("")
+                                                .to_string(),
+                                            _ => "".to_string(),
```

Comment:
For here you should be able to use `Path::extension` in some form like:
```rust
doc.path()
.and_then(|p| p.extension())
.and_then(|e| e.to_str())
.unwrap_or_default();
```
Leaves any wonkiness that could happen to the rust implimentation. I think what you have now works, but you never know.

Reply:
agreed, didn't know about this function, will update tomorrow